docker/Makefile

```diff
@@ -3,18 +3,77 @@ ROOT_IMAGE ?= alpine:3.12
 CERT_IMAGE := alpine:3.12
 GOLANG_IMAGE := golang:1.15-alpine
 
-BASE_IMAGE := localhost/baseimg:$(VERSION)-$(shell echo $(ROOT_IMAGE) | tr : -)
-DEBUG_IMAGE := localhost/debugimg:$(VERSION)-$(shell echo $(GOLANG_IMAGE) | tr : -)
+BASE_IMAGE := $(DOCKER_NAMESPACE)/baseimg:$(VERSION)-$(shell echo $(ROOT_IMAGE) | tr : -)
+DEBUG_IMAGE := $(DOCKER_NAMESPACE)/debugimg:$(VERSION)-$(shell echo $(GOLANG_IMAGE) | tr : -)
 
 create-baseimg-debugimg: create-baseimg create-debugimg
 
+dockerx-baseimg-debugimg: setup-buildx-builder
+	BUILDX_PUSH=true $(MAKE) create-baseimg-debugimg
+
 create-baseimg:
-	docker build -t $(BASE_IMAGE) \
+	$(DOCKER_BUILD) -t $(BASE_IMAGE) \
 		--build-arg root_image=$(ROOT_IMAGE) \
 		--build-arg cert_image=$(CERT_IMAGE) \
 		docker/base
 
 create-debugimg:
-	docker build -t $(DEBUG_IMAGE) \
+	$(DOCKER_BUILD) -t $(DEBUG_IMAGE) \
 		--build-arg golang_image=$(GOLANG_IMAGE) \
 		docker/debug
+
+# Ensure "docker buildx" is available and enabled. For more details, see: https://github.com/docker/buildx/blob/master/README.md
+# This does a few things:
+#  1. Makes sure docker is in PATH
+#  2. Downloads and installs buildx if no version of buildx is installed yet
+#  3. Makes sure any installed buildx is a required version or newer
+#  4. Makes sure the user has enabled buildx (either by default or by setting DOCKER_CLI_EXPERIMENTAL env var to 'enabled')
+#  Thus, this target will only ever succeed if a required (or newer) version of 'docker buildx' is available and enabled.
+install-docker-buildx-if-need:
+	@if ! which docker > /dev/null 2>&1; then echo "'docker' is not in your PATH."; exit 1; fi
```

Comment:
This should really be in an external shell script. In fact, I think it was there before, wasn't it?